sklearn/manifold/_t_sne.py

```diff
@@ -745,6 +747,28 @@ class TSNE(BaseEstimator):
     (4, 2)
     """
 
+    _parameter_constraints = {
+        "n_components": [Interval(Integral, 1, None, closed="left")],
+        "perplexity": [Real],
+        "early_exaggeration": [Interval(Real, 1, None, closed="left")],
+        "learning_rate": [
+            StrOptions({"auto", "warn"}),
+            Interval(Real, 0, None, closed="neither"),
+        ],
+        "n_iter": [Interval(Integral, 250, None, closed="left")],
+        "n_iter_without_progress": [Integral],
+        "min_grad_norm": [Real],
+        "metric": [StrOptions(set(_VALID_METRICS) | {"precomputed"}), callable],
+        "metric_params": [dict, None],
+        "init": [StrOptions({"pca", "random", "warn"}), np.ndarray],
+        "verbose": [Interval(Integral, 0, None, closed="left")],
+        "random_state": ["random_state"],
+        "method": [StrOptions({"barnes_hut", "exact"})],
+        "angle": [Interval(Real, 0, 1, closed="both")],
+        "n_jobs": [None, Integral],
+        "square_distances": [bool, StrOptions({"deprecated"})],
```

Comment:
We have a `"boolean"` helper constraint, and in this case the `"deprecated"` option should be hidden as well.
```suggestion
        "square_distances": ["boolean", Hidden(StrOptions({"deprecated"}))],
```